ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuationTest.kt

```diff
@@ -0,0 +1,172 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat.Companion.assertThatRule
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Test
+
+class ChainMethodContinuationTest {
+    private val chainMethodContinuationAssertThat = assertThatRule { ChainMethodContinuation() }
+
+    @Test
+    fun `Given method chain which is correctly wrapped in single line`() {
+        val code =
+            """
+            val foo = listOf(1, 2, 3).filter { it > 2 }!!.takeIf { it.count() > 100 }.map { it * it }?.sum()!!
+            """.trimIndent()
+        chainMethodContinuationAssertThat(code).hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given method chain which is incorrectly wrapped`() {
+        val code =
+            """
+            val foo = listOf(1, 2, 3).
+                filter { it > 2 }!!.
+                takeIf { it.count() > 100 }.
+                map {
+                    it * it
+                }?.
+                map {
+                    it * it
+                }!!.
+                sum()
+            """.trimIndent()
+        val formattedCode =
+            """
+            val foo = listOf(1, 2, 3)
+                .filter { it > 2 }!!
+                .takeIf { it.count() > 100 }
+                .map {
+                    it * it
+                }?.map {
+                    it * it
+                }!!.sum()
+            """.trimIndent()
+        chainMethodContinuationAssertThat(code)
+            .hasLintViolations(
+                LintViolation(1, 27, ". must merge at the start of next call"),
```

Comment:
What do you think about changing the message to `Unexpected newline after '.'`?

Reply:
What about `Expected newline before '.'` as someone can be confused the multiline chain is reported by the ktlint. Removing the newline after `.` will make the below code 
```
val foo = listOf(1, 2, 3).
                filter { it > 2 }!!.
```
to 
```
val foo = listOf(1, 2, 3).filter { it > 2 }!!.
```

Reply:
Although for cases like below
```
val foo = listOf(1, 2, 3).filter {
    true
}
.sum()
```
I am thinking of `Unexpected newline before '}.'` as here your suggestion seems more right to me

Reply:
Yeah, the message is depending on the situation. So, I am fine with both suggestions.